File: scramblies/javascript/1/scramblies.js
const a = 'a'.charCodeAt()

function scramble(str1, str2) {
  const len1 = str1.length
  const len2 = str2.length
  
  if (len1 < len2) {
    return false
  }
  
  const arr2 = Array(26).fill(0)
  for (let i = 0; i < len2; i++) {
    arr2[str2.charCodeAt(i) - a]++
  }
  
  for (let i = 0, j = len2; i < len1; i++) {
    const chr1 = str1.charCodeAt(i) - a
    const num2 = arr2[chr1]
    
    if (num2) {
      if (--j === 0) {
        return true
      }
      
      arr2[chr1] = num2 - 1
    }
  }
  
  return false
}

File: scramblies/java/1/Scramblies.java
public class Scramblies {
    public static boolean scramble(String str1, String str2) {
        int len1 = str1.length();
        int len2 = str2.length();

        if (len1 < len2) {
            return false;
        }

        int[] arr2 = new int['z' - 'a' + 1];
        for (int i = 0; i < len2; i++) {
            arr2[str2.charAt(i) - 'a']++;
        }

        for (int i = 0; i < len1; i++) {
            int chr1 = str1.charAt(i) - 'a';
            int num2 = arr2[chr1];

            if (num2 > 0) {
                if (--len2 == 0) {
                    return true;
                }

                arr2[chr1] = num2 - 1;
            }
        }

        return false;
    }
}